Feature/fidget updates

Jhonattan2121 opened the pull request:
# Changes & Pros

## Main Changes
- Converted Video.tsx to support Farcaster Frames
- Added unified style system (helpers.tsx)
- Implemented displayNameHint in all fidgets
- Standardized border/shadow/background settings

## Benefits
- Better integration with Farcaster
- More consistent UI/UX
- Clearer documentation for users
- Easier code maintenance

Jhonattan2121 committed:
fix: fixes application of heading fonts in widgets (#664)

Co-authored-by: Jesse Paterson <jesse.r.pat@gmail.com>

Jhonattan2121 committed:
Merge branch 'Nounspace:canary' into canary

Jhonattan2121 committed:
Merge branch 'Nounspace:canary' into canary

Jhonattan2121 committed:
Add instructions to each Fidget's settings

Jhonattan2121 committed:
Merge branch 'canary' into feature/fidget-updates

Jhonattan2121 committed:
fix video breaking on links

willyogo commented on src/fidgets/ui/Video.tsx:
```diff
@@ -1,18 +1,16 @@
-import useSafeUrl from "@/common/lib/hooks/useSafeUrl";
-import { defaultStyleFields } from "@/fidgets/helpers";
-import IFrameWidthSlider from "@/common/components/molecules/IframeScaleSlider";
-import { transformUrl } from "@/fidgets/helpers";
```
We need this transformUrl function to convert normal Youtube and Vimeo links into the embeddable format (ie. transforming https://www.youtube.com/watch?v=lOzCA7bZG_k to https://www.youtube.com/embed/lOzCA7bZG_k).

Please make sure users can still embed YouTube and Vimeo videos by pasting the normal URL

Jhonattan2121 committed:
video correction and regex

Jhonattan2121 committed:
fix: fixed Text.tsx style issues

willyogo commented on src/fidgets/ui/Text.tsx:
```diff
@@ -26,33 +26,44 @@ export type TextFidgetSettings = {
     },
     {
       fieldName: "fontFamily",
+      displayName: "FontFamily",
```
thanks for adding displayNames to these fields 🙏  while we're at it, please format any displayNames to use correct human syntax if they aren't already, ie. Font Family instead of FontFamily.

Jhonattan2121 committed:
fix: making it more humanized

Jhonattan2121 committed:
fix: title fix with text.tsx theme homepagetabsconfig.tsx

willyogo committed:
Update Feed.tsx

Updated and removed some unnecessary tooltips and changed 'the other app' to 'X'

willyogo committed:
Update PlatformSelector.tsx

Changing 'the other app' to 'X'

Jhonattan2121 committed:
Refactor fidget settings UI: simplify displayNameHint tooltips and standardize component spacing

Jhonattan2121 committed:
fix: word correction

Jhonattan2121 committed:
fix: word correction

Jhonattan2121 committed:
fix: input selector height correction

Jhonattan2121 committed:
refactoring: simplified Cast Fidget tooltips

willyogo committed:
Update Feed.tsx

willyogo committed:
Update Feed.tsx

willyogo committed:
Update Cast.tsx

willyogo committed:
Update Text.tsx

willyogo committed:
Update gallery.tsx

Jhonattan2121 committed:
fix: correction of tooltips and classes of music feed

Jhonattan2121 committed:
fix: image upload name correction

Jhonattan2121 committed:
fix: correction of fidget scalas

Jhonattan2121 committed:
fix: change following to corrected filter

Jhonattan2121 committed:
fix: resolve infinite render loop in Feed component using useRef for threadStack

j-paterson commented on src/common/components/organisms/TabBar.tsx:
```diff
@@ -175,6 +175,17 @@ function TabBar({
     }
   }
 
+   const handleTabClick = React.useCallback((tabName: string, e?: React.MouseEvent) => {
```
Is this just for debugging purposes?

Jhonattan2121 replied:
yes and also to make it work with it, you can make it work by clicking, for example, homebase profile, etc., before it wasn't working, now it's been fixed, it was being fixed with willy
